code-challenges.js

```diff
@@ -10,7 +10,48 @@ var people = [
 ]
 // Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is a president of the galaxy." "Arthus Dent is a radio employee."
 
+const sent = (obj) => {
 
+  //capture the length of the object being passed down in order to use in future loops
+  let length = obj.length
+
+  //capture all the "names" in the object
+  let arrNames = obj.map(value => value.name)
+
+  //seperate the first and last names 
+  let arrCap = arrNames.map(value => value.split(" "))
+
+  //once the first and last names are seperated we can now go in and captalize the first letter.
+  let name1 = arrCap.map(value => value[0].charAt(0).toUpperCase() + value[0].slice(1))
+  let name2 = arrCap.map(value => value[1].charAt(0).toUpperCase() + value[1].slice(1))
+
+  //once they are captilized we can now join them back together using cocat. now we have a names array that has the first and last names capatilzied
+  let names = name1.concat(name2)
+
+  //capture the occupations of the obj and put it into an array
+  let occ = obj.map(value => value.occupation)
+
+  //have an empty array ready to push in the first name last name and occupation all together into a sentence. By doing this we can create a sentence with every entry in the object
+  let finalArray = []
+
+  for (let i = 0; i < length; i++) {
+
+    // finalArray.push(`"${name1[i]} ${name2[i]} is a ${occ[i]}." `)
+    //inside of the for loop we now can compose a sentence using the names and occ array we created earlier that helped isolate the names to captialize and complete the sentence with the occ array we obtained
+    finalArray.push(`"${names[i]} is a ${occ[i]}." `)
+  }
+
+  //join it all together using join to make a complete sentence
+  return finalArray.join("")
+
+  // console.log(arrNames)
+  // console.log(arrCap)
+  // console.log(name1)
+  // console.log(name2)
+  // console.log(names)
+}
+
+console.log(sent(people))
```

Comment:
I think there's a tiny bug in the code here. Output is only give me `{name} is a {occupation}` not `{first name} {last name} is a {occupation}`